Modules/_json.c

```diff
@@ -1903,52 +1831,22 @@ encoder_listencode_dict(PyEncoderObject *s, PyUnicodeWriter *writer,
     return 0;
 
 bail:
+    Py_XDECREF(items);
     Py_XDECREF(ident);
     return -1;
 }
 
-static inline int
-_encoder_iterate_fast_seq_lock_held(PyEncoderObject *s, PyUnicodeWriter *writer,
-    PyObject *seq, PyObject *s_fast,
-    Py_ssize_t indent_level, PyObject *indent_cache, PyObject *separator)
-{
-    for (Py_ssize_t i = 0; i < PySequence_Fast_GET_SIZE(s_fast); i++) {
-        PyObject *obj = PySequence_Fast_GET_ITEM(s_fast, i);
-#ifdef Py_GIL_DISABLED
-        // gh-119438: in the free-threading build the critical section on s_fast can get suspended
-        Py_INCREF(obj);
-#endif
-        if (i) {
-            if (PyUnicodeWriter_WriteStr(writer, separator) < 0) {
-#ifdef Py_GIL_DISABLED
-                Py_DECREF(obj);
-#endif
-                return -1;
-            }
-        }
-        if (encoder_listencode_obj(s, writer, obj, indent_level, indent_cache)) {
-            _PyErr_FormatNote("when serializing %T item %zd", seq, i);
-#ifdef Py_GIL_DISABLED
-            Py_DECREF(obj);
-#endif
-            return -1;
-        }
-#ifdef Py_GIL_DISABLED
-        Py_DECREF(obj);
-#endif
-    }
-    return 0;
-}
-
 static int
 encoder_listencode_list(PyEncoderObject *s, PyUnicodeWriter *writer,
                         PyObject *seq,
                         Py_ssize_t indent_level, PyObject *indent_cache)
 {
     PyObject *ident = NULL;
     PyObject *s_fast = NULL;
+    Py_ssize_t i;
 
-    s_fast = PySequence_Fast(seq, "encoder_listencode_list needs a sequence");
+    ident = NULL;
+    s_fast = PySequence_Fast(seq, "_iterencode_list needs a sequence");
```

Comment:
Why is this?  I think it should be reverted, just as _encoder_iterate_fast_seq_lock_held() removal below.